src/matlab2tikz.m

```diff
@@ -5693,7 +5622,9 @@ function warnFacetedInterp(m2t)
         end
         table{iRow} = sprintf(FORMAT, thisData{:});
     end
-    table = lower(table); % convert NaN and Inf to lower case for TikZ
+    % convert NaN and Inf to lower case for TikZ
+    table = regexprep(table, '\<NaN\>', 'nan');
+    table = regexprep(table, '\<Inf\>', 'inf');
```

Comment:
Nice! I didn't even know about `\<` and `\>` in regular expressions. Thanks! I ❤️  the simplicity of this solution.